### PR TITLE
Add several more Kubernetes templates and audit rule remediations for the e8 and moderate profiles

### PR DESCRIPTION
**Pull request by jhrozek**
#### Description:

- Adds audit rule remediations for auditing of time-related syscalls
- Adds a Kubernetes template for generating audit rules that watch certain files
- Add a Kubernetes remediation that audits the clock_settime syscall

#### Rationale:

- We wanted to have the e8 profile covered for a demo this week, these patches
are leftovers I had in tree. They are also useful for the moderate profile
anyway.

### CODE REVIEW

**Comment by mildas** [comment]
**Changes identified:**
Others:
 Python abstract syntax tree change found in ssg/templates.py.

**Recommended tests to execute:**
 (cd build && cmake ../ && ctest -j4)

**Comment by JAORMX** [comment]
test this please

**Comment by mildas** [comment]
**Changes identified:**
Others:
 Python abstract syntax tree change found in ssg/templates.py.

**Recommended tests to execute:**
 (cd build && cmake ../ && ctest -j4)

File: docs/manual/developer_guide.adoc [hunk 1]
```diff
@@ -1488,6 +1488,12 @@ audit_file_contents::
 ** *contents* - expected contents of the file
 * Languages: Ansible, Bash, OVAL
 
+audit_rules_time::
+* Ensure there is an audit rule that records time-related syscalls. The syscall would be logged using the `audit_time_rules` key
+* Parameters:
+** *syscall* - name of the system call, e.g. `settimeofday`
+* Languages: Kubernetes
```
Comment (yuumasato) [inline docs/manual/developer_guide.adoc:1495]: Could the template be more generic (`audit_syscall_rule`) and the key parametrized?
There are a lot of audit rules about syscall that could benefit.

File: docs/manual/developer_guide.adoc [hunk 2]
```diff
@@ -1521,6 +1527,14 @@ audit_rules_usergroup_modification::
 ** *path* - path that should be part of the audit rule as a value of `-w` argument, eg. `/etc/group`.
 * Languages: Ansible, Bash, OVAL
 
+audit_add_watch_rule::
```
Comment (yuumasato) [inline docs/manual/developer_guide.adoc:1530]: Could you rename it to `audit_watch_rule`?

**Comment by jhrozek** [comment]
also todo: use rhcos4 content instead of ocp4

File: shared/templates/template_KUBERNETES_audit_add_watch_rule [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+kind: MachineConfig
+spec:
+  config:
+    ignition:
+      version: 2.2.0
+    storage:
+      files:
+      - contents:
+          source: data:,{{% for FILE in FILES -%}}
+          -w%20{{{ FILE }}}%20-p%20{{{ PERMISSIONS }}}%20-k%20{{{ KEY }}}%0A{{% endfor %}}
```
[the remark this answers is not in the document]
Comment (matejak) [reply shared/templates/template_KUBERNETES_audit_add_watch_rule:17]: You should be able to compose a human-readable string that you just pass to [`urlencode` filter](https://jinja.palletsprojects.com/en/2.11.x/templates/#urlencode) (unless https://github.com/pallets/jinja/issues/515 bothers you) or our extension, which is the `url_encode` macro that wraps `urllib.parse.quote`.

**Comment by matejak** [comment]
@jhrozek Will you be able to resume working in this PR? There is a minor doc conflict, and couple of comments from Yuuma that look reasonable in terms of implied work.

**Comment by mildas** [comment]
**Changes identified:**
Others:
 Python abstract syntax tree change found in ssg/templates.py.

**Recommended tests to execute:**
 (cd build && cmake ../ && ctest -j4)

**Comment by redhatrises** [comment]
Closing as the work on this PR has stalled and hasn't been worked on in more than a while.

**Pull request closed without merging**